src/main/java/usersandpostsCRUD/demo/service/CityService.java

```diff
@@ -0,0 +1,71 @@
+package usersandpostsCRUD.demo.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import usersandpostsCRUD.demo.dto.CityRequestBody;
+import usersandpostsCRUD.demo.dto.CityResponseBody;
+import usersandpostsCRUD.demo.entity.City;
+import usersandpostsCRUD.demo.entity.Country;
+import usersandpostsCRUD.demo.exception.CityNotFoundException;
+import usersandpostsCRUD.demo.repository.CityRepository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class CityService {
+
+    private final CityRepository cityRepository;
+    private final CountryService countryService;
+
+    @Autowired
+    public CityService(CityRepository cityRepository, CountryService countryService) {
+        this.cityRepository = cityRepository;
+        this.countryService = countryService;
+    }
+
+    public List<CityResponseBody> getAllCities() {
+        return cityRepository.findAll().stream()
+                .map(this::convertToResponseBody)
+                .collect(Collectors.toList());
+    }
+    private CityResponseBody convertToResponseBody(City city) {
+        return new CityResponseBody(city.getId(), city.getName(), city.getPostCode(), city.getCountry().getId());
+    }
+
+    public CityResponseBody getCityById(Long id) {
+        City city = cityRepository.findById(id)
+                .orElseThrow(() -> new CityNotFoundException(id));
+        return convertToResponseBody(city);
+    }
+
+    public CityResponseBody createCity(CityRequestBody cityRequestBody) {
+        Country country = countryService.findCountryById(cityRequestBody.getCountryId());
+        City city = new City();
```

Comment:
What if a city with a requestBody.getName() already exists in the database?

Is it possible to have country Macedonia and 2 cities Bitola?